packages/office-ui-fabric-react/src/components/Fabric/Fabric.base.tsx

```diff
@@ -47,13 +49,21 @@ export class FabricBase extends React.Component<
     const { as: Root = 'div', theme, dir } = this.props;
     const classNames = this._getClassNames();
     const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties, ['dir']);
-    const componentDir = getDir(theme, dir);
-    const parentDir = getDir(theme);
+    const getDir = getAllDir(theme, dir);
+    // If Fabric dir !== theme.RTL
+    // Or If theme.RTL !== page RTL
+    // Then we need to set dir on the Fabric component
+    const setComponentDir = getDir.component !== getDir.parent || getDir.component !== getDir.page;
```

Comment:
I'm super confused here. Some of the dir logic is in the `getAllDir` method, and some is here.

Can you just have a single method that gets the correct `dir` value and assigns it to the `dir` attribute?

```tsx
<Root dir={ getDir(theme, props) } ... />
```